dio/lib/src/utils.dart

```diff
@@ -30,17 +30,25 @@ String encodeMap(
   data,
   DioEncodeHandler handler, {
   bool encode = true,
+  bool isQuery = false,
   ListFormat listFormat = ListFormat.multi,
 }) {
   final urlData = StringBuffer('');
   bool first = true;
-  final leftBracket = encode ? '%5B' : '[';
-  final rightBracket = encode ? '%5D' : ']';
+  final leftBracket = !isQuery ? '%5B' : '[';
+  final rightBracket = !isQuery ? '%5D' : ']';
```

Comment:
Prefer not reverting the condition in usages.

```suggestion
  final leftBracket = isQuery ? '[' : '%5B';
  final rightBracket = isQuery ? ']' : '%5D';
```